src/pages/index.js

```diff
@@ -1,107 +1,244 @@
+import "../pages/index.css";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "55f75f9b-43cd-4f65-a141-e29dd49200ac",
+    "Content-Type": "application/json",
+  },
+});
 
 document.addEventListener("DOMContentLoaded", () => {
-  /* -------------------------- elements ----------------------------*/
   const profileEditButton = document.querySelector(".profile__edit-button");
   const addButton = document.querySelector("#profile-add-button");
   const nameInput = document.querySelector(".modal__input-title");
   const descriptionInput = document.querySelector(".modal__input-description");
   const profileEditForm = document.querySelector("#profile-form");
   const addCardForm = document.querySelector("#add-form");
 
-  /* ---------------------------- Instantiating Classes ----------------------------- */
-  // UserInfo class manages displaying and updating user profile info
   const userInfo = new UserInfo({
     nameSelector: ".profile__title",
     jobSelector: ".profile__description",
   });
 
-  // Section class handles rendering cards and adding them to the DOM
   const cardSection = new Section(
     {
-      items: initialCards,
+      items: [],
       renderer: (cardData) => {
-        const cardElement = createCard(cardData);
-        cardSection.addItem(cardElement); // Add card to the DOM
+        const cardElement = createCard(cardData, api); // Pass api to Card
+        cardSection.addItem(cardElement);
       },
     },
     ".cards__list-content"
   );
 
-  // PopupWithImage for displaying full-size image in a modal
   const popupWithImage = new PopupWithImage("#modal-image-preview");
   popupWithImage.setEventListeners();
 
-  // PopupWithForm for editing profile info
   const profileEditPopup = new PopupWithForm(
     "#profile-edit-modal",
     (formData) => {
-      userInfo.setUserInfo(formData); // Update the user info on the page
-      profileEditPopup.close(); // Close the modal after form submission
+      const updatedData = {
+        name: formData.name,
+        about: formData.job,
+      };
+
+      // Change the button text to "Saving..."
+      const saveButton = document.querySelector("#modal-button");
+      saveButton.textContent = "Saving...";
+
+      api
+        .updateProfile(updatedData)
+        .then((userData) => {
+          userInfo.setUserInfo({
+            name: userData.name,
+            job: userData.about,
+          });
+          const avatarImage = document.querySelector(".profile__image");
+          avatarImage.src = userData.avatar;
+          avatarImage.alt = userData.name;
+          profileEditPopup.close();
+        })
+        .catch((error) => {
+          console.error("Error updating profile information:", error);
+        })
+        .finally(() => {
+          // Reset the button text after the update (whether it was successful or not)
+          saveButton.textContent = "Save";
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```